### PR TITLE
new context manipulation ContextAssociationTransition

### PR DESCRIPTION
**Pull request by jannast**
new context manipulation ContextAssociationTransition

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

**Review by a-kleinf — APPROVED** [review]

File: src/t2iapi/context/context_requests.proto [hunk 1]
```diff
@@ -70,3 +70,15 @@ Request to associate a specific patient type.
 message AssociatePatientRequest {
     PatientType patient_type = 1;  // Patient type to be associated
 }
+
+/*
+Request to create or adapt a context state for the given descriptor handle with the starting @ContextAssociation value
+given as "start_association", to perform a transition to the final @ContextAssociation value, given as
+"end_association", happens during the remote procedure call.
```
Comment (ldeichmann) [inline src/t2iapi/context/context_requests.proto:77]: I don't understand the "happens during the remote procedure call.", that doesn't fit into the sentence. 
> to perform a transition to the final @ContextAssociation value [...] happens during the remote procedure call.

File: src/t2iapi/context/service.proto [hunk 2]
```diff
@@ -151,4 +151,12 @@ service ContextService {
    */
   rpc AssociatePatientWithAutoGeneratedPatientId (google.protobuf.Empty)
       returns (t2iapi.context.CreateContextStateWithAssociationResponse);
+  
+  /*
+  Create or adapt a context state for the given descriptor handle with the starting @ContextAssociation value
+  given as "start_association" (to trigger a context state report) and perform a transition of the previously
```
Comment (ldeichmann) [inline src/t2iapi/context/service.proto:157]: If the context is already in that association and is only being adapted, how does it trigger a report? This doesn't make sense.

**Review by ldeichmann — CHANGES_REQUESTED** [review]

File: src/t2iapi/context/context_requests.proto (hunk 1 above)
Comment (pstaeck) [reply src/t2iapi/context/context_requests.proto]: I removed it.

File: src/t2iapi/context/service.proto (hunk 2 above)
Comment (pstaeck) [reply src/t2iapi/context/service.proto]: done

File: src/t2iapi/context/service.proto [hunk 3]
```diff
@@ -151,4 +151,15 @@ service ContextService {
    */
   rpc AssociatePatientWithAutoGeneratedPatientId (google.protobuf.Empty)
       returns (t2iapi.context.CreateContextStateWithAssociationResponse);
+  
+  /*
+  For the pm:AbstractContextDescriptor with the provided handle create or adapt a pm:AbstractContextState
+  <state_of_interest> and ensure that the following sequence appears:
+  - State 1:
+     - <state_of_interest> has its @ContextAssociation value set the given "start_association"
```
Comment (ldeichmann) [inline src/t2iapi/context/service.proto:159]:
```suggestion
     - <state_of_interest> has its @ContextAssociation value set to the given "start_association"
```

File: src/t2iapi/context/service.proto [hunk 4]
```diff
@@ -151,4 +151,15 @@ service ContextService {
...
+  - State 2:
+     - <state_of_interest> has its @ContextAssociation value set the given "end_association"
```
Comment (ldeichmann) [inline src/t2iapi/context/service.proto:161]:
```suggestion
     - <state_of_interest> has its @ContextAssociation value set to the given "end_association"
```

**Review by ldeichmann — CHANGES_REQUESTED** [review]

File: src/t2iapi/context/service.proto (hunk 3 above)
Comment (pstaeck) [reply src/t2iapi/context/service.proto]: done

File: src/t2iapi/context/service.proto (hunk 4 above)
Comment (pstaeck) [reply src/t2iapi/context/service.proto]: done

**Review by ldeichmann — APPROVED** [review]

**Review by a-kleinf — APPROVED** [review]

**Comment by pstaeck** [comment]
since no test is using this manipulation anymore, the PR will be closed without merging

**Pull request closed without merging**